datastore/importers/nmdb-import-prowler/Parser.cpp

```diff
@@ -45,9 +45,13 @@ Parser::fromJsonV2(std::ifstream& _file)
   std::string line;
   while (std::getline(_file, line)) {
     json jline = json::parse(line);
-    nmdop::ProwlerV2Data v2d {jline};
-    if (v2d != nmdop::ProwlerV2Data()) {
-      d.v2Data.push_back(v2d);
+    if (!jline.empty()) {
+      nmdop::ProwlerV2Data v2d {jline};
+      if (v2d != nmdop::ProwlerV2Data()) {
+        d.v2Data.push_back(v2d);
+      }
+    } else {
+      std::cerr << "Malformed input: Empty JSON data." << std::endl;
     }
   }
   r.push_back(d);
```

Comment:
Given the issue context, this is a problematic line.  It will add something to `r` (the result) regardless of success or failure.  Basically, we should be checking to see if `d` is empty/default or not prior to adding to `r`.

FYSA: A similar concept can be seen in line 50-52.